### PR TITLE
DMP-2813: Adding endpoint to patch automated tasks

### PR DESCRIPTION
**Pull request by tonymort**
https://tools.hmcts.net/jira/browse/DMP-2813

 Adding endpoint to patch automated tasks

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/darts/task/service/AdminPatchAutomatedTaskTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    @Autowired
+    private GivenBuilder given;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @ParameterizedTest
+    @EnumSource(value = SecurityRoleEnum.class, names = {"SUPER_ADMIN"}, mode = Mode.INCLUDE)
+    void allowsSuperAdminToRetrieveAllAutomatedTasks(SecurityRoleEnum role) throws Exception {
```
Comment (chrisbellingham-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/darts/task/service/AdminPatchAutomatedTaskTest.java:34]: Test name copy/paste errror?

File: src/integrationTest/java/uk/gov/hmcts/darts/task/service/AdminUpdateAutomatedTaskServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    @Test
+    void updatesIsEnabledFieldTrue() {
+        var persistedAutomatedTaskEntity = dartsDatabase.getAutomatedTask(1);
+        final boolean initialValue = persistedAutomatedTaskEntity.getTaskEnabled(); // So we can revert it later as part of test clean up
+        persistedAutomatedTaskEntity.setTaskEnabled(false);
+        dartsDatabase.save(persistedAutomatedTaskEntity);
+
+        var resposne = adminAutomatedTaskService.updateAutomatedTask(1, new AutomatedTaskPatch().isActive(true));
```
Comment (jackmaloney) [inline src/integrationTest/java/uk/gov/hmcts/darts/task/service/AdminUpdateAutomatedTaskServiceTest.java:51]: typo in 'resposne'

File: src/integrationTest/java/uk/gov/hmcts/darts/task/service/AdminUpdateAutomatedTaskServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SuppressWarnings("PMD.ExcessiveImports")
+class AdminUpdateAutomatedTaskServiceTest extends IntegrationBase {
+
+    @Autowired
+    private AdminAutomatedTaskService adminAutomatedTaskService;
+
+    @Test
+    void updatesIsEnabledFieldFalse() {
```
Comment (jackmaloney) [inline src/integrationTest/java/uk/gov/hmcts/darts/task/service/AdminUpdateAutomatedTaskServiceTest.java:17]: What is the point of having 2 integration tests here? The assertions look to be testing the exact same thing except for the 'getTaskEnabled' check. It's not clear to me what we are trying to achieve here by having these in separate tests. Do we need both?

**Review by chrisbellingham-hmcts — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/darts/task/service/AdminUpdateAutomatedTaskServiceTest.java (hunk 3 above)
Comment (tonymort) [reply src/integrationTest/java/uk/gov/hmcts/darts/task/service/AdminUpdateAutomatedTaskServiceTest.java:17]: When theres just 2 possible results I just think may as well cover them both given its quick to write and run.  Can just have 1 but I think the test name should change to be more generic.

**Review by jackmaloney — APPROVED** [review]

**Pull request merged**